### PR TITLE
Fixes broken confirmation URLs for search Brave Ads

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/29419

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc [hunk 1]
```diff
@@ -54,15 +54,18 @@ mojom::UrlRequestInfoPtr CreateConfirmationUrlRequestBuilder::Build() {
...
   if (confirmation_.opted_in && confirmation_.opted_in->credential_base64url) {
     credential_base64_url =
         base::StrCat({"/", *confirmation_.opted_in->credential_base64url});
   }
 
   const std::string spec = base::StringPrintf(
-      "%s/v3/confirmation/%s%s",
-      GetAnonymousHost(confirmation_.ad_type).c_str(),
+      "%s/v3/confirmation/%s%s", host.c_str(),
       confirmation_.transaction_id.c_str(), credential_base64_url.c_str());
```
Comment (aseren) [inline components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc:69]: lets put DCHECK that confirmation_.transaction_id is not empty

File: components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc [hunk 2]
```diff
@@ -54,15 +54,18 @@ mojom::UrlRequestInfoPtr CreateConfirmationUrlRequestBuilder::Build() {
 ///////////////////////////////////////////////////////////////////////////////
 
 GURL CreateConfirmationUrlRequestBuilder::BuildUrl() const {
+  const std::string host = confirmation_.ad_type == AdType::kSearchResultAd
+                               ? server::GetAnonymousSearchHost()
+                               : server::GetAnonymousHost();
+
   std::string credential_base64_url;
   if (confirmation_.opted_in && confirmation_.opted_in->credential_base64url) {
     credential_base64_url =
```
Comment (aseren) [inline components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc:63]: DCHECK that  confirmation_.opted_in-> credential_base64_url is not empty

File: components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc [hunk 3]
```diff
@@ -54,15 +54,18 @@ mojom::UrlRequestInfoPtr CreateConfirmationUrlRequestBuilder::Build() {
...
         base::StrCat({"/", *confirmation_.opted_in->credential_base64url});
   }
 
   const std::string spec = base::StringPrintf(
-      "%s/v3/confirmation/%s%s",
-      GetAnonymousHost(confirmation_.ad_type).c_str(),
+      "%s/v3/confirmation/%s%s", host.c_str(),
       confirmation_.transaction_id.c_str(), credential_base64_url.c_str());
 
   return GURL(spec);
```
Comment (aseren) [inline components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc:71]: do we have a DCHECK that returning url is valid?

Comment (tmancey) [reply components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc:71]: Added a DCHECK to ads_service_impl

**Review by aseren — APPROVED** [review]

**Pull request merged**